Detectors/util/ecal_layer_stack.py

```diff
@@ -214,8 +446,33 @@ def print_weights(dE_between_sensdet, X0_between_sensdet, L_between_sensdet, Zpo
         'Back', dE_between_sensdet[-1], X0_between_sensdet[-1], L_between_sensdet[-1], Zpos_layer[-1]))
     output.flush()
 
-def main() :
-    # Section, bilayers, front, cooling
+
+def command(func):
+    command.__list__[func.__name__] = func
+    return func
+
+command.__list__ = {}
+
+
+@command
+def print_layer_materials():
+    print(json.dumps({
+        name : {
+            attr: getattr(material, attr)()
+            for attr in [
+                'minimum_ionization_MeV_mm',
+                'radiation_length_mm',
+                'nuclear_interaction_length_mm'
+            ]
+        }
+        for name, material in Layer.materials.items()
+    }, indent=2))
+
+
+@command
+def ldmx_ecal_v14():
+    """full LDMX Ecal v14 geometry"""
+    # section, bilayers, front, cooling
     absorber_sections = [
             ('a',1,1,1),
             ('b',1,2,1.5),
```

Comment:
Can you please add a bit more about what is happening here? is the 1.5 in mm? is this the thickness of the absorber in mm at a given location? What does it mean "cooling" of 1.5? Is that the thickness of the cooling? But that's not an "absorber section", I'm a little confused

Reply:
This is good motivation for this just to be a class rather than an undocumented tuple. I'll make that change and add documentation about what these numbers mean there.